test/new-e2e/tests/apm/docker_test.go

```diff
@@ -0,0 +1,159 @@
+			"datadog.trace_agent.trace_writer.bytes_uncompressed": {},
+			"datadog.trace_agent.trace_writer.errors":             {},
+			"datadog.trace_agent.trace_writer.events":             {},
+			"datadog.trace_agent.trace_writer.payloads":           {},
+			"datadog.trace_agent.trace_writer.retries":            {},
+			"datadog.trace_agent.trace_writer.spans":              {},
+			"datadog.trace_agent.trace_writer.traces":             {},
+		}
+		metrics, err := s.Env().FakeIntake.Client().GetMetricNames()
+		assert.NoError(c, err)
+		s.T().Log("Got metric names", metrics)
+		assert.GreaterOrEqual(c, len(metrics), len(expected))
+		for _, m := range metrics {
+			delete(expected, m)
+			if len(expected) == 0 {
+				s.T().Log("All expected metrics are found")
+				return
+			}
+		}
+		s.T().Log("Remaining metrics", expected)
+		assert.Empty(c, expected)
+	}, 2*time.Minute, 10*time.Second, "Failed finding datadog.trace_agent.* metrics")
+}
+
+func (s *DockerFakeintakeSuite) TracesOnUDS() {
+	run, rm := dockerRunUDS("tracegen-uds")
+	s.Env().Host.MustExecute(run)
+	defer s.Env().Host.MustExecute(rm)
+	s.EventuallyWithTf(func(c *assert.CollectT) {
+		traces, err := s.Env().FakeIntake.Client().GetTraces()
+		assert.NoError(c, err)
+		assert.NotEmpty(c, traces)
+		s.T().Log("Got traces", traces)
+		assert.True(c, hasContainerTag(traces, "container_name:tracegen-uds"))
+	}, 2*time.Minute, 10*time.Second, "Failed finding traces with container tags")
+}
+
+func (s *DockerFakeintakeSuite) StatsOnUDS() {
+	run, rm := dockerRunUDS("tracegen-stats-uds")
+	s.Env().Host.MustExecute(run)
+	defer s.Env().Host.MustExecute(rm)
+	s.EventuallyWithTf(func(c *assert.CollectT) {
+		stats, err := s.Env().FakeIntake.Client().GetAPMStats()
+		assert.NoError(c, err)
+		assert.NotEmpty(c, stats)
+		s.T().Log("Got apm stats", stats)
+		assert.True(c, hasStatsForService(stats, "tracegen-stats-uds"))
+	}, 2*time.Minute, 10*time.Second, "Failed finding stats")
+}
+
+func hasStatsForService(payloads []*aggregator.APMStatsPayload, service string) bool {
+	for _, p := range payloads {
+		for _, s := range p.StatsPayload.Stats {
+			for _, bucket := range s.Stats {
+				for _, ss := range bucket.Stats {
+					if ss.Service == service {
+						return true
+					}
+				}
+			}
+		}
+	}
+	return false
+}
+
+func hasContainerTag(payloads []*aggregator.TracePayload, tag string) bool {
+	for _, p := range payloads {
+		for _, t := range p.AgentPayload.TracerPayloads {
+			tags, ok := t.Tags["_dd.tags.container"]
+			if ok && strings.Count(tags, tag) > 0 {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func dockerRunUDS(service string) (string, string) {
+	// TODO: use a proper docker-compose definition for tracegen
+	run := "docker run -d --network host --rm --name " + service + " -v /var/run/datadog/:/var/run/datadog/ -e DD_TRACE_AGENT_URL=unix:///var/run/datadog/apm.socket -e DD_SERVICE=" + service + " ghcr.io/datadog/apps-tracegen:main"
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
This should be doable on the docker client, that uses docker sdk. We use it to run command on containers deployed within a docker compose https://github.com/DataDog/datadog-agent/blob/main/test/new-e2e/pkg/utils/e2e/client/docker.go#L56-L60

Reply:
Thanks! It looks like `ExecuteCommand` does the equivalent of `docker exec` in a given container, and cannot be used to run new containers which is what we're doing here.
My plan (as written in the TODO) is to define the fake workload in a docker compose and provide it with `WithExtraComposeManifest`. Lmk if it's a bad idea 😅

Reply:
We'll have to do it in the next task once we start tweaking the fake workload config.

Reply:
I actually like the inline method here, since it's super clear and in the same code as the test being run.